packages/core/image-asset/image-asset-common.ts

```
@@ -47,8 +47,30 @@ export function getAspectSafeDimensions(sourceWidth, sourceHeight, reqWidth, req
 export function getRequestedImageSize(src: { width: number; height: number }, options: ImageAssetOptions): { width: number; height: number } {
-	let reqWidth = options.width || Math.min(src.width, Screen.mainScreen.widthPixels);
-	let reqHeight = options.height || Math.min(src.height, Screen.mainScreen.heightPixels);
+	const optionsCopy = { ...(options || {}) };
+
+	if (typeof optionsCopy.width === 'string') {
+		const parsedWidth = parseInt(optionsCopy.width, 10);
+		if (!isNaN(parsedWidth)) {
+			optionsCopy.width = parsedWidth;
+		} else {
+			console.warn('Invalid width value provided: ', optionsCopy.width);
+			delete optionsCopy.width;
```

Biome (1.9.4)

[error] Avoid the delete operator which can impact performance.

Unsafe fix: Use an undefined assignment instead.


(lint/performance/noDelete)